### ISSUE TITLE
nextPrevFilter endTime issue if time interval filter used within visualization

### ISSUE THREAD

**Issue by killanch** [nuagenetworks/vis-graphs#254]
If `Time interval` filter is used within the visualization, then endTime value does not get reset on reselecting `Time interval` filter value.

Sample:
![nextprev](https://user-images.githubusercontent.com/24920919/66787433-8f696700-ee98-11e9-96a5-f755718448a1.gif)

In the above example, on re-selecting the last 15 min filter option, the end time being used corresponds to the last set value by NextPrevFilter component.

Failed query where end time value used is lesser than start time:
```
{
  "size":0,
  "query":{
    "bool":{
      "must":[
        {
          "range":{
            "timestamp":{
              "gte":"now-15m",
              "lte":"now-45m",
              "format":"epoch_millis"
            }
          }
        },
        {
          "term":{
            "system_id":"45.205.62.163"
          }
        }
      ]
    }
  },
  "aggs":{
    "ts":{
      "date_histogram":{
        "field":"timestamp",
        "interval":"1m",
        "extended_bounds":{
          "min":"now-15m",
          "max":"now-45m"
        }
      },
      "aggs":{
        "CPU":{
          "avg":{
            "field":"cpu"
          }
        },
        "MEMORY":{
          "avg":{
            "field":"memory"
          }
        }
      }
    }
  }
}
```

The query fails with below error:
```
{
  "error":{
    "root_cause":[
      {
        "type":"search_parse_exception",
        "reason":"[extended_bounds.min][1571091604092] cannot be greater than [extended_bounds.max][1571089804092] for histogram aggregation [ts]"
      }
    ],
    "type":"search_phase_execution_exception",
    "reason":"all shards failed",
    "phase":"query",
    "grouped":true,
    "failed_shards":[
      {
        "shard":0,
        "index":"nuage_sysmon-2019-10-14",
        "node":"gZMZYkBuQk-jIggXEyKxVQ",
        "reason":{
          "type":"search_parse_exception",
          "reason":"[extended_bounds.min][1571091604092] cannot be greater than [extended_bounds.max][1571089804092] for histogram aggregation [ts]"
        }
      }
    ]
  },
  "status":400
}
```

This issue is NOT seen if the time interval filter is at the dashboard level.

**Comment by killanch** [comment]
@nxanil can you pls have a look?
cc @natabal

**Comment by natabal** [comment]
@nxanil Let's fix this one! Thanks

**Comment by killanch** [comment]
@nxanil the below issues still need to be addressed,

But I noticed other related issues.
**1.** I chose `Last 1h`, scrolled to previous window, all was good.
When I come back to original window, though data is rendered correctly, the filter value gets reset to `Last 15 min` and not the original chosen value `Last 1h`.
So the data rendered and the filter value displayed do not correspond to each other.
Can we fix this as well?

![nextprevfilter2](https://user-images.githubusercontent.com/24920919/67028282-e309c980-f0bf-11e9-91ff-bb4cf7bef4a5.gif)

**2.** Minor issue related to style. When we use the prev filter and move to previous window, the `Time Interval` filter position gets reset and moves all the way to the right of the card. Position should be maintained. This is visible in the above screen capture as well.
